pkg/service/service.go

```
@@ -1005,10 +1050,7 @@ func (s *Service) deleteServiceLocked(svc *svcInfo) error {
 	for _, id := range obsoleteBackendIDs {
 		scopedLog.WithField(logfields.BackendID, id).
 			Debug("Deleting obsolete backend")
-
-		if err := s.lbmap.DeleteBackendByID(id, ipv6); err != nil {
-			return err
-		}
+		s.lbmap.DeleteBackendByID(id)
```

Why ignoring errors?

---

The id is unique across both v4/v6 backends. Only based on the ID we don't know whether a backend is in the v4 or v6 map, so we need to attempt to delete it from both. Also, when we remove orphan backends in general I don't think we should bail from that with error.

---

If we store the NAT policy in `svcInfo`, then we could be able to determine which backends need exactly to be deleted.

If someone extends `DeleteBackendByID()`, then the err ignore might get into our way. Usually when I see a function returning an error, my expectation that the error return is going to be handled by callers.

---

For the case of `deleteOrphanBackends` right now it's actually worse. If for some reason the backend map returns `ENOENT` if the to-be-deleted backend doesn't exist anymore in the BPF map (non-fatal error, and the only one BPF map can return), then we cancel all remaining deletion (of unrelated backends) and leaving stale backends with refcount 0 intact. Caller won't handle anything, just dumping a warn log message.